### PR TITLE
Complete `Ephemeral.vue` component for problems page

### PR DESCRIPTION
**Pull request by HNOONa-0**
# Description

I have added logic for both submitting and running a problem. The component is now ready to be used instead of the `ephemeral.ts` script for the problems page case.

https://github.com/user-attachments/assets/c361f42c-0def-4e2b-a7fb-388beef84134

Here is also a video showing that it works with a contest:

https://github.com/user-attachments/assets/351010dd-78bc-4e68-bd21-2f2ddb7784f1

There will be another separate pull request for full screen IDE.
Fixes: #6214, fixes: #7761, fixes: #7749

# Comments
There are multiple files that are not used any longer like `config-grader`. These will be addressed when completing the component for full screen IDE.
I modified some existing tests to suit this component.
# Checklist:

- [x] The code follows the [coding guidelines](https://github.com/omegaup/omegaup/wiki/Coding-guidelines) of omegaUp.
- [x] The tests were executed and all of them passed.
- [x] If the change is large (> 200 lines), this PR was split into various Pull Requests. It's preferred to create one PR for changes in controllers + unit tests in PHPUnit,  and then another Pull Request for UI + tests in Jest, Cypress or both.

### CODE REVIEW

**Comment by HNOONa-0** [comment]
@pabo99 PTAL!

File: frontend/www/js/omegaup/components/arena/EphemeralGrader.vue [hunk 1]
```diff
@@ -1,27 +1,29 @@
...
+  },
+})
 export default class EphemeralGrader extends Vue {
-  @Ref() grader!: HTMLIFrameElement;
   @Prop() problem!: types.ProblemInfo;
   @Prop({ default: false }) canSubmit!: boolean;
   @Prop({ default: true }) canRun!: boolean;
   @Prop({ default: () => [] }) acceptedLanguages!: string[];
-  @Prop({ default: 'cpp17-gcc' }) preferredLanguage!: string;
+  @Prop({ default: 'cpp17gcc' }) preferredLanguage!: string;
```
Comment (pabo99) [inline frontend/www/js/omegaup/components/arena/EphemeralGrader.vue:26]: Is this change intentional? If so, what is the reason?

File: frontend/www/js/omegaup/graderv2/Ephemeral.vue [hunk 2]
```diff
@@ -126,48 +143,215 @@ export default class Ephemeral extends Vue {
...
+        if (!response.ok) return null;
+
+        // refreshing appends a hash to the current url at the end
+        // i dont think this is intended behaviour
+
+        // history.replaceState(
+        // undefined
+        //   '',
+        //   '#' + response.headers.get('X-OmegaUp-EphemeralToken'),
+        // );
```
Comment (pabo99) [inline frontend/www/js/omegaup/graderv2/Ephemeral.vue:294]: The comment seems legit, could you remove all the commented lines please?

File: frontend/www/js/omegaup/graderv2/MonacoEditor.vue [hunk 3]
```diff
@@ -17,10 +17,15 @@ export default class MonacoEditor extends Vue {
   };
   @Prop({ default: 'vs-dark' }) theme!: string;
   @Prop({ default: false }) readOnly!: boolean;
+  @Ref('monaco-div') readonly monacoDiv!: HTMLElement;
```
Comment (pabo99) [inline frontend/www/js/omegaup/graderv2/MonacoEditor.vue:20]: Is this reference still necessary? If not, is it possible to remove it?

**Review by pabo99 — COMMENTED** [review]
Excellent job! You are close to finishing the migration!

One more thing, is it possible to add more tests? This will help avoid Codecov complaints in the GitHub Action checks.

File: frontend/www/js/omegaup/components/arena/EphemeralGrader.vue (hunk 1 above)
Comment (HNOONa-0) [reply frontend/www/js/omegaup/components/arena/EphemeralGrader.vue]: That was not intentional.  I reverted that change.

**Comment by HNOONa-0** [comment]
> Excellent job! You are close to finishing the migration!
> 
> One more thing, is it possible to add more tests? This will help avoid Codecov complaints in the GitHub Action checks.

Yes but I felt like it would be too hard to review. I could modify the tests that I turned off.

**Review by pabo99 — COMMENTED** [review]
Thank you @HNOONa-0, It LGTM

Could you please skip the flaky test in Cypress? I have tried unsuccessfully to run the GitHub Actions to pass. So, for now, we can omit it in order to approve this PR

@heduenas,  can you help us with a final review of this PR? This way Ahmed could continue working with the standalone version of the Ephemeral component

**Comment by HNOONa-0** [comment]
> Thank you @HNOONa-0, It LGTM
> 
> Could you please skip the flaky test in Cypress? I have tried unsuccessfully to run the GitHub Actions to pass. So, for now, we can omit it in order to approve this PR
> 
> @heduenas, can you help us with a final review of this PR? This way Ahmed could continue working with the standalone version of the Ephemeral component

I think I found out why its flaky, there is more than one class of `CodeMirror-lines` (three in this case) and only one is visible. We can fix this.
image

**Comment by HNOONa-0** [comment]
> > > Thank you @HNOONa-0, It LGTM
> > > Could you please skip the flaky test in Cypress? I have tried unsuccessfully to run the GitHub Actions to pass. So, for now, we can omit it in order to approve this PR
> > > @heduenas, can you help us with a final review of this PR? This way Ahmed could continue working with the standalone version of the Ephemeral component
> > 
> > 
> > I think I found out why its flaky, there is more than one class of `CodeMirror-lines` (three in this case) and only one is visible. We can fix this. image
> 
> It would be great! Feel free to fix it in a new PR; it can help to get it approved quickly.

Got it! I will disable it for now.

**Review by pabo99 — APPROVED** [review]
Excellent!

File: cypress/e2e/course.cy.ts [hunk 4]
```diff
@@ -244,7 +238,7 @@ describe('Course Test', () => {
     cy.logout();
   });
 
-  it('Should create a course and leave feedback for the submission', () => {
+  xit('Should create a course and leave feedback for the submission', () => {
```
Comment (heduenas) [inline cypress/e2e/course.cy.ts:241]: why are you disabling this test?

Comment (heduenas) [reply cypress/e2e/course.cy.ts:241]: I guess you only meant to disable it to test locally without submitting, right?

Comment (HNOONa-0) [reply cypress/e2e/course.cy.ts]: This is a flaky test, @pabo99 suggested to disable it until a separate pull request is made to fix it.

Comment (pabo99) [reply cypress/e2e/course.cy.ts]: Actually, I had asked Ahmed to skip the test because it was failing in all attempts to run the GitHub actions. It’s okay that you have re-enabled it. In any case, the Cypress tests are not mandatory to approve the PR.

**Review by heduenas — APPROVED** [review]

File: cypress/e2e/course.cy.ts (hunk 4 above)
Comment (HNOONa-0) [reply cypress/e2e/course.cy.ts]: > Actually, I had asked Ahmed to skip the test because it was failing in all attempts to run the GitHub actions. It’s okay that you have re-enabled it. In any case, the Cypress tests are not mandatory to approve the PR.

Hmmm, I thought this is how you skip a test, you mean `it.skip` is the correct syntax?

Comment (pabo99) [reply cypress/e2e/course.cy.ts]: Both functions do the same thing, so it doesn’t matter which one we use. What we should do in another PR is try to fix that flaky test.
